markwon-image-coil/src/main/java/io/noties/markwon/image/coil/CoilImagesPlugin.java

```diff
@@ -0,0 +1,186 @@
+package io.noties.markwon.image.coil;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.text.Spanned;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.commonmark.node.Image;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import coil.Coil;
+import coil.ImageLoader;
+import coil.api.ImageLoaders;
+import coil.request.LoadRequest;
+import coil.request.RequestDisposable;
+import coil.target.Target;
+import io.noties.markwon.AbstractMarkwonPlugin;
+import io.noties.markwon.MarkwonConfiguration;
+import io.noties.markwon.MarkwonSpansFactory;
+import io.noties.markwon.image.AsyncDrawable;
+import io.noties.markwon.image.AsyncDrawableLoader;
+import io.noties.markwon.image.AsyncDrawableScheduler;
+import io.noties.markwon.image.DrawableUtils;
+import io.noties.markwon.image.ImageSpanFactory;
+
+/**
+ * @since 4.0.0
```

Comment:
You can put here some attribution if you wish, like your name and email, but it's optional